common/sphinx.c

```diff
@@ -89,10 +89,14 @@ struct sphinx_path *sphinx_path_new_with_key(const tal_t *ctx,
 
 static size_t sphinx_hop_count_frames(const struct sphinx_hop *hop)
 {
+#if EXPERIMENTAL_FEATURES
```

Comment:
This is likely redundant with the check on line 426, since this just clamps the number of frames to 1, never triggering the failure, and probably corrupting the frame if we end up using more anyway. I suggest we just keep the check below, and restore the normal behavior.

Reply:
OK.